File: Day-1/two_sum.py
def twoSum(nums, target):
    n = len(nums)
    arr = []

    for i in range(n):
        arr.append((nums[i], i))

    arr.sort()

    i, j = 0, n - 1

    while i < j:
        ps = arr[i][0] + arr[j][0]

        if ps == target:
            return [arr[i][1], arr[j][1]]
        elif ps < target:
            i += 1
        else:
            j -= 1

    return []


if __name__ == "__main__":
    nums = [2, 7, 11, 15]
    target = 9

    result = twoSum(nums, target)

    if result:
        print("Indices:", result)
    else:
        print("No solution found")

File: Day-1/two_sum.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    int n = nums.size();
    vector<pair<int,int>> arr;

    for(int i = 0; i < n; i++) {
        arr.push_back({nums[i], i});
    }

    sort(arr.begin(), arr.end());

    int i = 0, j = n - 1;

    while(i < j)
    {
        int PS = arr[i].first + arr[j].first;

        if(PS == target)
        {
            return {arr[i].second, arr[j].second};
        }
        else if(PS < target)
        {
            i++;
        }
        else
        {
            j--;
        }
    }
    return {};
}

int main() {
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;

    vector<int> result = twoSum(nums, target);

    if(!result.empty()) {
        cout << "Indices: " << result[0] << " " << result[1] << endl;
    } else {
        cout << "No solution found" << endl;
    }

    return 0;
}